src/wui/productionsitewindow.cc

```diff
@@ -261,16 +261,18 @@ void ProductionSiteWindow::evict_worker() {
 	}
 
 	if (worker_table_->has_selection()) {
-		Widelands::Worker* worker = production_site->working_positions()
-		                               ->at(worker_table_->get_selected())
-		                               .worker.get(ibase()->egbase());
-		if (worker != nullptr) {
-			if (game_ != nullptr) {
-				game_->send_player_evict_worker(*worker);
-			} else {
-				NEVER_HERE();  // TODO(Nordfriese / Scenario Editor): implement
-			}
-		}
+	  worker_table_->select(worker_table_->get_selected());
```

Comment:
This line is redundant, you can collapse the if-else into a shorter `if (!worker_table_->has_selection())`